src/test/java/com/datadog/api/v2/client/api/SecurityMonitoringApiTest.java

```diff
@@ -242,71 +242,103 @@ public void searchSignals() throws Exception {
+            } catch (ApiException ignored) {
+                return false;
+            }
+        });
 
         // Paging
-        SecurityMonitoringSignalsListResponse pageOneResponse = api.listSecurityMonitoringSignals()
-                .filterQuery(uniqueName)
-                .filterFrom(now.minus(Duration.ofHours(1)))
-                .filterTo(now.plus(Duration.ofHours(1)))
-                .pageLimit(1)
-                .execute();
+        TestUtils.retry(5, 10, () -> {
+            try {
+                // First page
+                SecurityMonitoringSignalsListResponse pageOneResponse = api.listSecurityMonitoringSignals()
+                        .filterQuery(uniqueName)
+                        .filterFrom(now.minus(Duration.ofHours(1)))
+                        .filterTo(now.plus(Duration.ofHours(1)))
+                        .pageLimit(1)
+                        .execute();
 
-        assertEquals(1, pageOneResponse.getData().size());
+                if (pageOneResponse.getData() == null || pageOneResponse.getData().size() < 1)
+                    return false;
 
-        String cursor = pageOneResponse.getMeta().getPage().getAfter();
-        assertTrue(pageOneResponse.getLinks().getNext().contains(URLEncoder.encode(cursor)));
+                String cursor = pageOneResponse.getMeta().getPage().getAfter();
+                boolean correctFirstLink = pageOneResponse.getLinks().getNext().contains(URLEncoder.encode(cursor));
 
-        SecurityMonitoringSignalsListResponse pageTwoResponse = api.listSecurityMonitoringSignals()
-                .filterQuery(uniqueName)
-                .filterFrom(now.minus(Duration.ofHours(1)))
-                .filterTo(now.plus(Duration.ofHours(1)))
-                .pageLimit(1)
-                .pageCursor(cursor)
-                .execute();
+                // Second page
+                SecurityMonitoringSignalsListResponse pageTwoResponse = api.listSecurityMonitoringSignals()
+                        .filterQuery(uniqueName)
+                        .filterFrom(now.minus(Duration.ofHours(1)))
+                        .filterTo(now.plus(Duration.ofHours(1)))
+                        .pageLimit(1)
+                        .pageCursor(cursor)
+                        .execute();
 
-        assertEquals(1, pageTwoResponse.getData().size());
+                if (pageTwoResponse.getData() == null || pageTwoResponse.getData().size() < 1)
+                    return false;
 
-        assertNotEquals(pageOneResponse.getData().get(0).getId(), pageTwoResponse.getData().get(0).getId());
+                boolean correctSecondLink = !pageOneResponse.getData().get(0).getId().equals(pageTwoResponse.getData().get(0).getId());
+                return correctFirstLink && correctSecondLink;
+            } catch (ApiException ignored) {
+                return false;
+            }
+        });
 
         api.deleteSecurityMonitoringRule(rule.getId());
```

Comment:
If the test fails, we won't do this clean up step. Not sure how we can do a "finally" in Junit, besides @After

Reply:
The normal pattern is 
```
try {
} finally {
  // cleanup
}
```
`@After ` might be sufficient though